src/main.rs

```diff
@@ -93,7 +95,12 @@ async fn main() -> ExitCode {
         a library is mis-using the tracing API.",
     );
 
-    match provision().await {
+    let opts = Cli::parse();
+
+    let config = Config::load(opts.config.clone())
+        .expect("Failed to load configuration");
```

Comment:
Instead of panicking here, we could print out the error and an example correct config:

```Rust
    let config = Config::load(opts.config.clone())
        .map_err(|error| {
            eprintln!("Failed to load configuration ({error:?})")
            eprintln!("Example configuration:\n\n{}", Config::default());
        });
```
And then return `ExitCode::FAILURE` if load return an `Err`. This way the user gets a non-zero exit code (which they'd also get with a panic) but don't get a potentially confusing backtrace. Incorrect configuration is a "normal" thing, and we don't want to crash ungracefully unless we're in a state we didn't expect.

Since we don't know what log level the user might have configured, we print directly to standard error. You may also need to implement `Display` for `Config` to make the above work, something like (totally untested):
```
impl Display for Config {
    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
        write!(
            f,
            "{}",
            toml::ser::to_string_pretty(&self).unwrap_or_default()
        )
    }
}
```